doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -476,14 +476,28 @@ In Ltac2, we have the following backtracking primitives, defined in the
   val plus : (unit -> 'a) -> (exn -> 'a) -> 'a
   val case : (unit -> 'a) -> ('a * (exn -> 'a)) result
 
-If one views thunks as lazy lists, then `zero` is the empty list and `plus` is
-list concatenation, while `case` is pattern-matching.
+Informally, the primitive `zero e` ends the current computation and returns to the last backtracking point with exception `e`.
+The primitive `plus t handler` installs a backtracking point around `t`, using `handler` to handle any exception raised by a `zero` while evaluating `t`.
+The primitive `case t` inspects the backtracking structure of a thunked computation `t` and returns `Err e` if `t` yields `zero e`, or returns the first backtracking point as a pair `Val(v,h)` of a value `v` and a handler `h` if there is any. In the particular case where the computation `t ()` yields an immediate value `v`, `case t` returns a trivial backtracking point `Val(v, zero)`.
+
+From the point of view of backtracking, thunks of type `unit -> 'a` (corresponding to suspended computations of type `'a`) could be modeled as lazy lists holding values of type `'a` where `zero` corresponds to the empty list and `plus` to list concatenation, while `case` is pattern-matching. To take exceptions into account, the empty list is parameterized by an exception, while the tail of a non-empty list is thunked by an exception (since `(exn -> 'a) ≅ (exn -> unit -> 'a)`).
```

Comment:
I'd still leave this paragraph before the introduction of the combinators, I think it's clearer to say out loud first that thunks are streams.

Reply:
Here's a new version with the backtracking model coming first. I deferred the description of the API to come just after the model.